### PR TITLE
Task/download fragment crash fix

### PR DESCRIPTION
**Pull request by CalebKL**
Currently WIP
Fixes https://github.com/kiwix/kiwix-android/issues/3949

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt [hunk 1]
```diff
@@ -288,7 +297,13 @@ class ZimManageViewModel @Inject constructor(
     Function6(::combineLibrarySources)
   )
     .doOnNext { libraryListIsRefreshing.postValue(false) }
+    .doOnError { throwable ->
+      if (throwable is OutOfMemoryError) {
+        Log.e("ZimManageViewModel", "Error----${throwable.printStackTrace()}")
+      }
+    }
     .subscribeOn(Schedulers.io())
+    .observeOn(AndroidSchedulers.mainThread())
```
Comment (MohitMaliFtechiz) [inline app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:306]: @CalebKL Why we are now starting to observe this on the main thread?

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt (hunk 1 above)
Comment (CalebKL) [reply app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:306]: I have made the changes to remove this line

Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:306]: But I am still seeing this code.

**Review by MohitMaliFtechiz — APPROVED** [review]
@CalebKL Thanks :tada: :tada:

**Comment by MohitMaliFtechiz** [comment]
@CalebKL Can you please resolve the conflicts.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 LGTM.

**Pull request merged**